src/docs/asciidoc/Batch-Jobs.asciidoc

```diff
@@ -725,6 +726,66 @@ job create sqoopComplexArgs1 --definition "sqoop --command=import --args='--tabl
 
 (For this example we have omitted the equal sign for the individual Sqoop arguments within the `--args` option. Either style works fine.)
 
+===== Using Sqoop's metastore
+
+It is possible to use Sqoop's metastore with some restrictions.
+
+WARNING: Sqoop ships with HSQLDB version 1.8 and Spring XD ships with HSQLDB version 2.3. Since these two versions are not compatible you can not use a Sqoop metastore
+that uses HSQLDB. This is unfortunate since HSQLDB version 1.8 is the only database that is fully supported for the metastore by Sqoop. We can however use another database
+for the metastore as long as we use some workarounds.
+
+NOTE: You can use PostgreSQL for the Sqoop metastore. We recommend that you run the commands listed below to create and initialize the tables to be used by the Sqoop metastore.
+
+Create and initialize the Sqoop metastore tables:
+
+----
+CREATE TABLE
+    SQOOP_ROOT
+    (
+        version INTEGER,
+        propname CHARACTER VARYING(128) NOT NULL,
+        propval CHARACTER VARYING(256),
+        UNIQUE (version, propname)
+    );
+CREATE TABLE
+    SQOOP_SESSIONS
+    (
+        job_name CHARACTER VARYING(64) NOT NULL,
+        propname CHARACTER VARYING(128) NOT NULL,
+        propval CHARACTER VARYING(1024),
+        propclass CHARACTER VARYING(32) NOT NULL,
+        UNIQUE (job_name, propname, propclass)
+    );
+INSERT INTO sqoop_root (version, propname, propval) VALUES (null, 'sqoop.hsqldb.job.storage.version', '0');
+INSERT INTO sqoop_root (version, propname, propval) VALUES (0, 'sqoop.hsqldb.job.info.table', 'SQOOP_SESSIONS');
+----
+
+You can now modify the `scoop-site.xml` file in the Spring XD config directory. Add the JDBC URL, username and password to use for connection to the PostgreSQL database
+that hosts the Sqoop metastore tables. You need to provide the following properties:
+
+- `sqoop.metastore.client.autoconnect.url`
+- `sqoop.metastore.client.autoconnect.username`
+- `sqoop.metastore.client.autoconnect.password`
+
+NOTE: In addition to the above configurations you need to use a `--password-file` option when creating the Sqoop job definitions. If you don't then Sqoop will prompt for a password
+as Spring XD runs the job. This will cause the job to hang.
+
+Here is an example of defining a Sqoop job using Spring XD's `sqoop` job:
+
+----
+xd>job create job1create --definition "sqoop --command=job --args='--create job1 -- import --table PETS --incremental append --check-column ID --last-value 0 --connect jdbc:hsqldb:hsql://localhost:9001/test --username sa --password-file /xd/hsql.password --target-dir /xd/job1 --num-mappers 1'" --deploy
```

Comment:
I used gpdb as db. Below is a command what sqoop would run when job is later executed. It fails if table is in caps and later we get same trouble with column name.

```
create table PETS (id integer, name text) distributed by (id);

gpadmin=# SELECT t.* FROM "PETS" AS t LIMIT 1;
ERROR:  relation "PETS" does not exist
LINE 1: SELECT t.* FROM "PETS" AS t LIMIT 1;
                        ^
gpadmin=# SELECT t.* FROM "pets" AS t LIMIT 1;
 id | name 
----+------
  4 | pet4

gpadmin=# select count(*) from pets;
 count 
-------
     5
(1 row)

gpadmin=# select count(*) from PETS;
 count 
-------
     5
(1 row)

gpadmin=# select count(*) from "PETS";
ERROR:  relation "PETS" does not exist
LINE 1: select count(*) from "PETS";
                             ^
gpadmin=# select count(*) from "pets";
 count 
-------
     5
(1 row)
```
Anyway just adding note here.